research/2d-decoding/rectangle_detector.py

```diff
@@ -212,14 +231,18 @@ def readImage(filename):
     print("Plotting")
     for rect in rects:
         for i in range(4):
-            plt.plot(*rect[i], marker="o", markersize=3, markeredgecolor="red", markerfacecolor="red")
+            plt.plot(*rect[i], marker="o", markersize=3,
+                     markeredgecolor="red", markerfacecolor="red")
         print("Rect done")
     # Flip the y axis so that the origin is in the top left
     plt.savefig("processed.png")
     # clear the plot
     plt.clf()
     print("Done")
-        
+
+    # Return the rectangles as a list of lists of tuples
```

Comment:
isn't it a list of tuples?

Reply:
There is a list of quadrilaterals. Each quadrilateral is a list of coordinates. Each coordinate is a tuple. Comment added 👍